src/storage/mvcc/reader/reader.rs

```diff
@@ -185,15 +218,36 @@ impl<S: EngineSnapshot> MvccReader<S> {
         Ok(res)
     }
 
-    fn load_in_memory_pessimistic_lock(&self, key: &Key) -> Option<Lock> {
-        self.snapshot.ext().get_txn_ext().and_then(|txn_ext| {
-            txn_ext
-                .pessimistic_locks
-                .read()
-                .map
-                .get(key)
-                .map(|lock| lock.to_lock())
-        })
+    fn load_in_memory_pessimistic_lock(&self, key: &Key) -> Result<Option<Lock>> {
+        self.snapshot
+            .ext()
+            .get_txn_ext()
+            .and_then(|txn_ext| {
+                // If the term or region version has changed, do not read the lock table.
```

Comment:
I'm not sure if it's necessary to check it, but it's safer.

Reply:
It seems not necessary to check the version. I add both checks to become safer and consistent.

Reply:
I need to comment out the version check part. Because we haven't maintained it well. Then, it cause unavailabilities after split and merge, even when this feature is not enabled.
The term is always updated when becoming leader, so I think the current code is safe.